src/identity/configuration/IdentityProviderFactory.ts

```diff
@@ -316,16 +340,37 @@ export class IdentityProviderFactory implements ProviderFactory {
    * Pipes library errors to the provided ErrorHandler and ResponseWriter.
    */
   private configureErrors(config: Configuration): void {
-    config.renderError = async(ctx: KoaContextWithOIDC, out: ErrorOut, error: Error): Promise<void> => {
+    config.renderError = async(ctx: KoaContextWithOIDC, out: ErrorOut, error: errors.OIDCProviderError | Error):
+    Promise<void> => {
       // This allows us to stream directly to the response object, see https://github.com/koajs/koa/issues/944
       ctx.respond = false;
 
-      // OIDC library hides extra details in this field
-      if (out.error_description) {
-        error.message += ` - ${out.error_description}`;
+      // Doesn't really matter which type it is since all relevant fields are optional
+      const oidcError = error as errors.OIDCProviderError;
+
+      // OIDC library hides extra details in these fields
+      if (this.showStackTrace) {
+        if (oidcError.error_description) {
+          error.message += ` - ${oidcError.error_description}`;
+        }
+        if (oidcError.error_detail) {
+          oidcError.message += ` - ${oidcError.error_detail}`;
+        }
+
+        // Also change the error message in the stack trace
+        if (error.stack) {
+          error.stack = [ `${error.name}: ${error.message}`, ...error.stack.split('\n').slice(1) ].join('\n');
```

Comment:
Why not just ``error.stack.replace(/.*/, `${error.name}: ${error.message}`)``?

Reply:
Because my regex-fu is simply not strong enough.